File: app/src/main/java/com/example/attendance/fragments/snapmit/submit/SubmitViewModel.java
package com.example.attendance.fragments.snapmit.submit;

import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

import java.io.File;
import java.util.ArrayList;
import java.util.HashMap;
import java.util.Map;

public class SubmitViewModel extends ViewModel {
    MutableLiveData<ArrayList<File>> imagesData;
    MutableLiveData<Map<String, Integer>> versionMap;
    MutableLiveData<String> code;


    public SubmitViewModel() {
        imagesData = new MutableLiveData<>();
        imagesData.setValue(new ArrayList<>());
        versionMap = new MutableLiveData<>();
        versionMap.setValue(new HashMap<>());
        code = new MutableLiveData<>();
        code.setValue("");
    }

}

File: app/src/main/java/com/example/attendance/viewmodels/SubmitViewModel.kt
package com.example.attendance.viewmodels

import androidx.lifecycle.ViewModel
import com.example.attendance.util.android.SafeLiveData
import java.io.File

class SubmitViewModel : ViewModel() {
    var imagesData: SafeLiveData<List<File>> = SafeLiveData(arrayListOf())
    var versionMap: SafeLiveData<Map<String, Int>> = SafeLiveData(mutableMapOf())
    var assignmentUUID: SafeLiveData<String> = SafeLiveData("")
}
